### ISSUE TITLE
Ansible being required for non-remote delegates

### ISSUE THREAD

**Issue by fubarhouse** [fubarhouse/ansible-role-tester#48]
See https://github.com/issmirnov/ansible-role-compile-zsh/pull/3

Ansible is causing a potential problem for logging where it's not being explicitly required for localhost delegates when being applied on the remote.

If you don't declare remote behaviour being desired, this message simply should not be shown.

Evidence of this feature can be presented via: https://github.com/fubarhouse/ansible-role-tester/blob/master/util/util.go#L95-L99

This actually presents undesired behaviour downstream and should be treated as a priority.

**Comment by fubarhouse** [comment]
@issmirnov,

Could I ask that you test branch `issue-48` if and when you get an opportunity?

It will mean that Ansible won't be required unless the tool is used with Ansible delegates and not running the commands inside of Docker containers.

**Comment by issmirnov** [comment]
Hmm, it's crashing for me.

```
# install
go get github.com/fubarhouse/ansible-role-tester
cd ~/go/src/github.com/fubarhouse/ansible-role-tester 
git checkout issue-48
git rev-parse HEAD # aca46c3e41af60f31228c5fd8c4c0a1020f1b061
go install .


cd ~/ansible-role-tester

# run 1
ansible-role-tester full -v 
ERRO[0000] exit status 1  

# run 2
ansible-role-tester full --destination /etc/ansible/roles/$(basename $(pwd)) --playbook tests/playbook.yml
ERRO[0000] exit status 1 

# run 3
ansible-role-tester run -t ubuntu1804 -name "test"
ERRO[0000] exit status 1    

```

**Comment by fubarhouse** [comment]
Alright thanks, I'll make a dedicated effort to test it somewhere without Ansible because I'm not seeing that :(

**Comment by issmirnov** [comment]
@fubarhouse 

I tested on an alternative system and it works. I think this might be issues with my docker installation.

**Comment by issmirnov** [comment]
Looks like the error log I saw was due to my user not being in the `docker` group, and thus the line at https://github.com/fubarhouse/ansible-role-tester/blob/master/util/docker.go#L136 was triggered. 

One low priority suggestion is to add a small check to see if the docker daemon is reachable, and notify the user if not. This would also be encountered if someone has a CI build where the user running `ansible-role-tester` doesn't have docker privileges.

**Comment by issmirnov** [comment]
Since you don't want to import the entire docker library, I would say simplest method that comes to mind is to try and dial the socket directly.

In docker, they have [this](https://github.com/docker/docker-ce/blob/6e92e5909666b3b9c2aecebf582e8af85f228899/components/engine/client/hijack.go#L61) check. 

Something like 
```
c, err := net.Dial("unix", "/var/run/docker.sock")
if err != nil {
    og.Fatal("Unable to connect to docker", err)
}
defer c.Close()
```

What do you think?

**Comment by fubarhouse** [comment]
Sounds like a case of close enough is good enough. It's almost completely what I had in mind when I last thought about it.

**Comment by fubarhouse** [comment]
It feels good to be able to code again on a personal device!

Those commits appear to be passing now, and I've reintroduced the check against the Docker daemon. It's almost midnight here so I'll wait till morning to check the outcome and ask for feedback.

I see no roadblocks in getting a PR made and/or merged to master.

**Comment by fubarhouse** [comment]
Now waiting on feedback to submit a PR.

**Comment by issmirnov** [comment]
Looks neat! I don't know what IDE you use, and if to doesn't auto format you may wish to run `gofmt -s -w .` for the report card rating - it'll autoformat and write the changes.

I tested the branch a few days ago and it worked perfectly. I'll try to find time today to check the docker logic on my laptop, but I'm sure you tested it as well.

Excited for this to land! I'm always amazed at your quick responses on this project 
- thank you!

**Comment by fubarhouse** [comment]
I'll configure it to use gofmt soon - it's GoLand.
I've been using it since it was announced on Christmas Eve 2016 👍 

I'll organise a PR after getting some formatting fixes in place.

**Comment by issmirnov** [comment]
Update: yep, it detects when docker is down. Nice!

**Comment by fubarhouse** [comment]
PR is ready for merging, if you've nothing more to add in the next day I'll merge it tomorrow.

**Comment by issmirnov** [comment]
LGTM :)

**Comment by fubarhouse** [comment]
Merged!
